src/index.ts

```diff
@@ -1,94 +1,78 @@
+    const appName = getInput('application-name', {required: true})
+    const groupName = getInput('deployment-group-name', {required: true})
+    const appspecFile = getInput('appspec-file', {required: true})
+    debug(`Hello world! ${appName}, ${groupName}, ${appspecFile}`)
 
-    const appspecJson = fs.readFileSync(appspecFile, 'utf8')
+    const appspecJson = readFileSync(appspecFile, 'utf8')
 
-    core.debug('*** appspecJson ***')
-    core.debug(appspecJson)
-    core.debug('*** end appspecJson ***')
+    debug('*** appspecJson ***')
+    debug(appspecJson)
+    debug('*** end appspecJson ***')
 
-    const codeDeploy = new CodeDeploy()
-    const deployment = await codeDeploy
-      .createDeployment({
-        applicationName: appName,
-        deploymentGroupName: groupName,
-        revision: {
-          revisionType: 'AppSpecContent',
-          appSpecContent: {
-            content: appspecJson
-          }
+    const codedeployClient = new CodeDeployClient()
+    const { deploymentId } = await codedeployClient.send(new CreateDeploymentCommand({
+      applicationName: appName,
+      deploymentGroupName: groupName,
+      revision: {
+        revisionType: 'AppSpecContent',
+        appSpecContent: {
+          content: appspecJson
         }
-      })
-      .promise()
-    core.debug(`deployment: ${JSON.stringify(deployment)}`)
+      }
+    }))
 
-    if (deployment.deploymentId == null) {
-      core.setFailed('deploymentId should not be null')
+    if (deploymentId == null) {
+      setFailed('deploymentId should not be null')
       return
     }
 
-    const {awsAccountAlias, awsAccountId} = await getAccountInformation()
-    const region = codeDeploy.config.region
+    const { awsAccountAlias, awsAccountId } = await getAccountInformation()
+    const region = codedeployClient.config.region
     const iamRole = `PowerUser-${awsAccountId}`
-    const destinationUrl = `https://${region}.console.aws.amazon.com/codesuite/codedeploy/deployments/${deployment.deploymentId}?region=${region}`
+    const destinationUrl = `https://${region}.console.aws.amazon.com/codesuite/codedeploy/deployments/${deploymentId}?region=${region}`
     const shortcutLink = `https://byulogin.awsapps.com/start/#/console?account_id=${encodeURIComponent(awsAccountId)}&role_name=${encodeURIComponent(iamRole)}&destination=${encodeURIComponent(destinationUrl)}`
-    core.info(`Started deployment.
+    info(`Started deployment.
     
-Deployment ID:    ${deployment.deploymentId}
+Deployment ID:    ${deploymentId}
 AWS Account:      ${awsAccountAlias} (${awsAccountId})
 Region:           ${region}
 
 Link to deployment: ${shortcutLink}`)
 
-    await codeDeploy
-      .waitFor('deploymentSuccessful', {
-        deploymentId: deployment.deploymentId
-      })
-      .promise()
+    const maxWaitTime = 3600 // One hour
+    await waitUntilDeploymentSuccessful({ client: codedeployClient, maxWaitTime }, { deploymentId })
```

Comment:
This was interesting. The new version had `maxWaitTime` as a required field. I believe v2 defaulted to 1 hour, so I used that.

Reply:
Edit: We now allow users to pass in a value, but still default to one hour.